### ISSUE TITLE
Creating a handful (~3) of rooms, for example to populate a new Space in Element Web, leads to running into rate limits

### ISSUE THREAD

**Issue by reivilibre** [matrix-org/synapse#14312]
With default rate limiting values, we used to be able to create at least 3 rooms in one go. The new Space screen in Element Web depends on this (https://github.com/vector-im/element-web/issues/23620) and breaks if any of those room creations are rate limited, as they are now (v1.70.0).

The following test case demonstrates this:

```python
from http import HTTPStatus

from synapse.rest.client import room
from tests.unittest import HomeserverTestCase, override_config


class BisectCase(HomeserverTestCase):
    servlets = [room.register_servlets]
    hijack_auth = True
    user_id = "@sid1:test"

    @override_config({
        "rc_message": {"per_second": 0.2, "burst_count": 10},
    })
    def test_can_create_3_rooms(self):
        """
        Should be able to create 3 rooms in a row (e.g. rooms in a space)
        without hitting rate limits.
        """

        for _ in range(3):
            channel = self.make_request(
                "POST",
                "/createRoom",
                {},
            )
            self.assertEqual(channel.code, HTTPStatus.OK, channel.json_body)
```

Bisecting, the following commit is what made the above fail because of the rate limit: 8ab16a92edd675453c78cfd9974081e374b0f998 'Persist CreateRoom events to DB in a batch (#13800)' in v1.69.0.

We should try to revert rate limiting to behave the same as before or closely enough that this basic use case is not broken.

**Comment by clokep** [comment]
Is #14307 attempting to fix this?

**Comment by reivilibre** [comment]
> Is #14307 attempting to fix this?

I don't know, but it doesn't pass

**Comment by reivilibre** [comment]
These places contribute to the rate limiting in the above test:

```
	~~~~~ Ratelimited actions:

	  File "/home/rei/work/synapse/synapse/http/server.py", line 229, in wrapped_async_request_handler
	    await h(self, request)
	  File "/home/rei/work/synapse/synapse/http/server.py", line 307, in _async_render_wrapper
	    callback_return = await self._async_render(request)
	  File "/home/rei/work/synapse/synapse/http/server.py", line 513, in _async_render
	    callback_return = await raw_callback_return
	  File "/home/rei/work/synapse/synapse/rest/client/room.py", line 161, in on_POST
	    info, _ = await self._room_creation_handler.create_room(
	  File "/home/rei/work/synapse/synapse/handlers/room.py", line 756, in create_room
	    await self.request_ratelimiter.ratelimit(requester)
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1487, in _persist_events
	    event = await self.persist_and_notify_client_events(
	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1649, in persist_and_notify_client_events
	    await self.request_ratelimiter.ratelimit(
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1487, in _persist_events
	    event = await self.persist_and_notify_client_events(
	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1649, in persist_and_notify_client_events
	    await self.request_ratelimiter.ratelimit(
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1487, in _persist_events
	    event = await self.persist_and_notify_client_events(
	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1649, in persist_and_notify_client_events
	    await self.request_ratelimiter.ratelimit(
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1487, in _persist_events
	    event = await self.persist_and_notify_client_events(
	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1649, in persist_and_notify_client_events
	    await self.request_ratelimiter.ratelimit(
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/http/server.py", line 229, in wrapped_async_request_handler
	    await h(self, request)
	  File "/home/rei/work/synapse/synapse/http/server.py", line 307, in _async_render_wrapper
	    callback_return = await self._async_render(request)
	  File "/home/rei/work/synapse/synapse/http/server.py", line 513, in _async_render
	    callback_return = await raw_callback_return
	  File "/home/rei/work/synapse/synapse/rest/client/room.py", line 161, in on_POST
	    info, _ = await self._room_creation_handler.create_room(
	  File "/home/rei/work/synapse/synapse/handlers/room.py", line 756, in create_room
	    await self.request_ratelimiter.ratelimit(requester)
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1487, in _persist_events
	    event = await self.persist_and_notify_client_events(
	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1649, in persist_and_notify_client_events
	    await self.request_ratelimiter.ratelimit(
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1487, in _persist_events
	    event = await self.persist_and_notify_client_events(
	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1649, in persist_and_notify_client_events
	    await self.request_ratelimiter.ratelimit(
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1487, in _persist_events
	    event = await self.persist_and_notify_client_events(
	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1649, in persist_and_notify_client_events
	    await self.request_ratelimiter.ratelimit(
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1487, in _persist_events
	    event = await self.persist_and_notify_client_events(
	  File "/home/rei/work/synapse/synapse/handlers/message.py", line 1649, in persist_and_notify_client_events
	    await self.request_ratelimiter.ratelimit(
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~+~

	  File "/home/rei/work/synapse/synapse/http/server.py", line 229, in wrapped_async_request_handler
	    await h(self, request)
	  File "/home/rei/work/synapse/synapse/http/server.py", line 307, in _async_render_wrapper
	    callback_return = await self._async_render(request)
	  File "/home/rei/work/synapse/synapse/http/server.py", line 513, in _async_render
	    callback_return = await raw_callback_return
	  File "/home/rei/work/synapse/synapse/rest/client/room.py", line 161, in on_POST
	    info, _ = await self._room_creation_handler.create_room(
	  File "/home/rei/work/synapse/synapse/handlers/room.py", line 756, in create_room
	    await self.request_ratelimiter.ratelimit(requester)
	  File "/home/rei/work/synapse/synapse/api/ratelimiting.py", line 391, in ratelimit
	    tb = "".join(traceback.format_stack())
	
	~~~~~
```

One might argue that we shouldn't ratelimit the creation of events whilst creating a room at all, because then you easily end up with a partially-created room which is very janky. I'd be tempted to rate limit the room's creation but not the events that go inside (?)

**Comment by clokep** [comment]
> One might argue that we shouldn't ratelimit the creation of events whilst creating a room at all [..] I'd be tempted to rate limit the room's creation but not the events that go inside

I think this likely makes sense.

**Comment by clokep** [comment]
Fixed by #14314.
